src/jobs/extraction_jobs.py

```diff
@@ -0,0 +1,237 @@
+            }
+
+        if not filename:
+            filename = "document.pdf"
+
+        # Convert extraction config dict to ExtractionConfig if provided
+        config = None
+        if extraction_config:
+            try:
+                config = ExtractionConfig(**extraction_config)
+            except Exception as e:
+                # Log warning but continue with defaults
+                config = None
+
+        # Extract markdown using existing logic
+        markdown, metadata = extract_markdown_with_hierarchy(
+            pdf_bytes, filename, config=config
+        )
+
+        # Calculate total processing time
+        total_processing_time = time.time() - job_start_time
+
+        # Enhance metadata with job information
+        enhanced_metadata = {
+            "filename": filename,
+            "processing_time": total_processing_time,
+            **metadata  # Include all extraction metadata
+        }
+
+        # Add analysis metadata if provided
+        if analysis_metadata:
+            enhanced_metadata["analysis_results"] = analysis_metadata
+
+        # Return result following rq_pymupdf pattern
+        return {
+            "ok": True,
+            "markdown": markdown,
+            "metadata": enhanced_metadata,
+            "filename": filename,
+            "processing_time": total_processing_time
+        }
+
+    except Exception as e:
+        # Handle any errors
+        return {
+            "ok": False,
+            "error": str(e),
+            "filename": filename,
+            "processing_time": time.time() - job_start_time
+        }
+
+
+def extract_pdf_with_config_job(
+    pdf_bytes: bytes,
+    filename: str,
+    margins: Optional[Tuple[int, int, int, int]] = None,
+    header_max_levels: Optional[int] = None,
+    header_body_limit: Optional[int] = None,
+    write_markdown: bool = False,
+    output_dir: Optional[str] = None,
+    analysis_metadata: Optional[Dict[str, Any]] = None
+) -> Dict[str, Any]:
+    """
+    RQ job for PDF extraction with detailed configuration options.
+
+    This job provides more granular control over extraction parameters
+    compared to the basic extraction job.
+    """
+    # Build extraction config from parameters
+    config_dict = {}
+
+    if margins:
+        config_dict["margins"] = margins
+
+    if header_max_levels is not None:
+        config_dict["header_detection_max_levels"] = header_max_levels
+
+    if header_body_limit is not None:
+        config_dict["header_detection_body_limit"] = header_body_limit
+
+    if write_markdown and output_dir:
+        config_dict["write_dir"] = output_dir
+        config_dict["write_mode"] = "overwrite"
+
+    # Use the main extraction job with custom config
+    return extract_pdf_markdown_job(
+        pdf_bytes=pdf_bytes,
+        filename=filename,
+        analysis_metadata=analysis_metadata,
+        extraction_config=config_dict
+    )
+
+
+def batch_extract_pdfs_job(
```

Comment:
We don't need to implement batching of job processing yet. Each `extract_pdf_markdown_job` only need to process 1 PDF at a time, and we simply need to scale up the number of worker processes to achieve parallelism